app/src/main/java/com/babylon/wallet/android/presentation/main/MainViewModel.kt

```diff
@@ -119,6 +121,10 @@ class MainViewModel @Inject constructor(
             }.collect()
         }
         handleAllIncomingRequests()
+
+        viewModelScope.launch {
+            syncAccountsWithConnectorExtensionUseCase()
```

Comment:
maybe here you can exploit the `hasAtLeastOneConnection` of the `PeerdroidClient` and start the syncing when you have at least one connection?

Reply:
This is already done within the use case, maybe I have to find a better name for the use case to reflect that

Reply:
exactly... this is what i was thinking. To not even check this in the usecase.